internal/app/app.go

```diff
@@ -0,0 +1,123 @@
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+)
+
+// set of exported json options
+const (
+	ExportedJSONPrefix = ""
+	ExportedJSONIndent = "    "
+)
+
+// Data is the exported Realm app config data which it is identified by
+type Data struct {
+	ID   string `json:"client_app_id,omitempty"`
+	Name string `json:"name"`
+}
+
+func (d Data) IsZero() bool {
+	return d.Name == ""
+}
+
+// Config is the exported Realm app config
+type Config struct {
+	Data
+	ConfigVersion        realm.AppConfigVersion `json:"config_version"`
+	Location             realm.Location         `json:"location"`
+	DeploymentModel      realm.DeploymentModel  `json:"deployment_model"`
+	Security             SecurityConfig         `json:"security"`
+	CustomUserDataConfig CustomUserDataConfig   `json:"custom_user_data_config"`
+	Sync                 SyncConfig             `json:"sync"`
+}
+
+// SecurityConfig ia the Realm app security config
+type SecurityConfig struct{}
+
+// CustomUserDataConfig is the Realm app custom user data config
+type CustomUserDataConfig struct {
+	Enabled bool `json:"enabled"`
+}
+
+// SyncConfig is the Realm app sync config
+type SyncConfig struct {
+	DevelopmentModeEnabled bool `json:"development_mode_enabled"`
+}
+
+// set of app structure filepath parts
+const (
+	FileConfig = "config.json"
+
+	DirAuthProviders = "auth_providers"
+
+	DirGraphQL                = "graphql"
+	DirGraphQLCustomResolvers = DirGraphQL + "/custom_resolvers"
+	FileGraphQLConfig         = DirGraphQL + "/config.json"
+)
+
+// FileAuthProvider creates the auth provider config filepath
+func FileAuthProvider(name string) string {
+	return fmt.Sprintf("%s/%s.json", DirAuthProviders, name)
+}
+
+var (
+	maxDirectoryContainSearchDepth = 8
+)
```

Comment:
this was mostly carry-over from the previous repo, and it's a `var` so we can make it a smaller value (2) in a test (making it _slightly_ less verbose to write)

we can make it a const, it would just mean updating a test line to be:

```go
superNestedRoot := filepath.Join(nestedRoot, "l4", "l5", "l6", "l7", "l8", "l9")
```